bigchaindb-server/bigchaindb/common/asset.py

```diff
@@ -0,0 +1,47 @@
+import signal
+
+
+class TimedOutExc(Exception):
+    pass
+
+
+def handler(signum, frame):
+    raise TimedOutExc()
+
+
+def validate_asset(transaction, bigchain):
+    try:
+        if not hasattr(transaction, 'asset'):
+            raise KeyError('asset not found in transaction {}'.format(transaction))
+
+        asset = transaction.asset
+        if 'id' in asset:
+            create_tx, _ = bigchain. \
+                get_transaction(asset['id'], include_status=True)
+            asset = create_tx.asset
+
+        if 'data' in asset:
+            asset_data = asset['data']
+            if 'script' in asset_data:
```

Comment:
If there is no 'script' in data, the conditions need to be bypassed and return True.
Else, BDB will not work for non-script transactions.